srcpkgs/Slic3r/template

```diff
@@ -0,0 +1,121 @@
+short_desc="Open Source toolpath generator for 3D printers"
+maintainer="Jasper Chan <jasperchan515@gmail.com>"
+license=GPL-3.0
+homepage="https://github.com/slic3r/Slic3r/"
+distfiles="https://github.com/slic3r/Slic3r/archive/1.3.0.tar.gz>slic3r.tar.gz"
+checksum=4fb351c1ae25c213cc6f17022eb62e2f0ce9678c0439916cb536d5cb3503da31
+
+pre_configure() {
+    # MakeMaker fails 2 tests every time
+    cpanm --force ExtUtils::MakeMaker
+
+    # Slic3r tests fail without local::lib
+    cpanm --force local::lib
+
+}
+
+#do_configure builds the backend
+
+do_build() {
+    # Socket fails 2 tests every time
+    cpanm --force Socket
+
+    # Wx tests must be run inside an X server
+    cpanm --force Wx
+    cpanm --force Wx::GLCanvas
+
+    # build the gui
+    perl Build.PL --gui
+}
+
+# Takes a folder and installs everything inside of it
+_recursive_vinstall() {
+    for file in "$1"/*; do
+        echo "Handling $file"
+        if [ -d "$file" ]; then
+            echo "$file is a directory, recursing..."
+            vmkdir "$3"/"$(basename $file)"
+            _recursive_vinstall "$1"/"$(basename $file)" "$2" "$3"/"$(basename $file)"
+        elif [ -f "$file" ]; then
+            echo "Installing $file to $3"
+            vinstall "$file" "$2" "$3"
+        fi
+    done
+}
+
+do_install() {
+    # ./Build script isn't found by do_install,
+    # forcing it to run by linking fails with
+    # ld: unrecognized option '-Wl,-z,-relro'
+    #
+    # Instead, I'm basing this script on the aur pkgbuild file
+    # https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=slic3r
+
+    echo "Dest dir:"
+    echo "$DESTDIR"
+    echo "pkgdest dir:"
+    echo "$PKGDESTDIR"
+
+    vmkdir /usr/lib/perl5/vendor_perl/
+    # This script should go into /usr/bin/
+    #vinstall slic3r.pl 755 /usr/lib/perl5/vendor_perl/
+
+    # Install utils, this is skipped in the aur pkgbuild file and I'm not sure why
+    vmkdir /usr/bin/
+    for file in utils/*.pl; do
+        vinstall "$file" 755 /usr/bin/
+    done
+
+    for file in utils/post-processing/*.pl; do
+        vinstall "$file" 755 /usr/bin/
+    done
+    for file in "$DESTDIR"/usr/bin/*.pl; do
+        rename -- .pl '' "$file"
+    done
+
+    # ZSH autocompletion
+    vmkdir /usr/share/zsh/site-functions/
+    vinstall utils/zsh/functions/_slic3r 644 /usr/share/zsh/site-functions/
+
+    # Icons " current Build.PL is not really geared for installation "
+    vmkdir /usr/bin/var/
+    for file in var/*; do
+        echo "Installing $file to vendor_perl/var"
+        vinstall "$file" 644 /usr/bin/var/
+    done
+
+    # Desktop icon
+    vmkdir /usr/share/applications/
+    vinstall package/linux/slic3r.desktop.in 644 /usr/share/applications/ slic3r.desktop
+
+    # Here the aur pkgbuild file just runs the Build script inside xs, but again that
+    # fails so I'm going to try and install things manually
+
+    vinstall slic3r.pl 755 /usr/bin/ Slic3r
+
+    _recursive_vinstall lib/ 755 /usr/lib/perl5/vendor_perl/
```

Comment:
wouldn't vcopy also do the job?

Reply:
vcopy doesnt seem to allow setting permissions